systemFlake.nix

```diff
@@ -86,24 +86,51 @@ let
     let
       selectedNixpkgs = getNixpkgs host;
       host = evalHostArgs (mergeAny hostDefaults host');
+      channelValue = channels.${host.channelName};
+      patchedChannel = patchChannel host.system channelValue.input (channelValue.patches or [ ]);
+      # Use lib from patched nixpkgs
+      lib = import (patchedChannel + "/lib");
+      # Use nixos modules from patched nixpkgs
+      baseModules = import (patchedChannel + "/nixos/modules/module-list.nix");
+      # Override `modulesPath` because otherwise imports from there will not use patched nixpkgs
+      specialArgs = { modulesPath = builtins.toString (patchedChannel + "/nixos/modules"); } // host.specialArgs;
```

Comment:
I believe using something along the lines of `patchedChannel = selectedNixpkgs.path;` should get the job done nicely.
Same goes for lib - `lib = selectedNixpkgs.lib;`
As for `baseModules` - I bet there is some way to do it as well, need to look into it. I will get back to you on this one.

Reply:
I was thinking of a possible alternative to this could be if you set the builder to `import "${selectedNixpkgs}/nixos/lib/eval-config.nix"`. That would only deal with the modules, but it wouldn't help with the nixpkgs.config issue.

lib.nixosSystem is essentially that but it sets the nixos revision and adds some vm features. So if we are willing to drop those, and just directly import eval-config, it might be much cleaner.

Reply:
Or maybe we can keep the default as nixosSystem, and just add documentation on how to pass the patched channel eval-config as the builder.

Reply:
Ah, I didn't know the path attribute existed. I'll make the requested changes. I'm pretty sure the base modules will always have to be imported, they aren't exposed in nixpkgs' pkgs.